src/Runner.Worker/FileCommandManager.cs

```diff
@@ -264,65 +317,161 @@ public void ProcessCommand(IExecutionContext context, string filePath, Container
         }
     }
 
-    public sealed class CreateStepSummaryCommand : RunnerService, IFileCommandExtension
+    public sealed class SetOutputFileCommand : RunnerService, IFileCommandExtension
     {
-        public const int AttachmentSizeLimit = 1024 * 1024;
-
-        public string ContextName => "step_summary";
-        public string FilePrefix => "step_summary_";
+        public string ContextName => "output";
+        public string FilePrefix => "set_output_";
 
         public Type ExtensionType => typeof(IFileCommandExtension);
 
         public void ProcessCommand(IExecutionContext context, string filePath, ContainerInfo container)
         {
-            if (String.IsNullOrEmpty(filePath) || !File.Exists(filePath))
-            {
-                Trace.Info($"Step Summary file ({filePath}) does not exist; skipping attachment upload");
-                return;
-            }
-
             try
             {
-                var fileSize = new FileInfo(filePath).Length;
-                if (fileSize == 0)
+                var pairs = new EnvFileKeyValuePairs(filePath);
+                foreach (var pair in pairs)
                 {
-                    Trace.Info($"Step Summary file ({filePath}) is empty; skipping attachment upload");
-                    return;
+                    context.SetOutput(pair.Key, pair.Value, out var reference);
+                    context.Debug($"Set output {pair.Key} = {pair.Value}");
                 }
+            }
+            catch (DirectoryNotFoundException)
+            {
+                context.Debug($"State file does not exist '{filePath}'");
+            }
+            catch (FileNotFoundException)
+            {
+                context.Debug($"State file does not exist '{filePath}'");
+            }
+        }
+    }
 
-                if (fileSize > AttachmentSizeLimit)
+    public sealed class EnvFileKeyValuePairs: IEnumerable<KeyValuePair<string, string>>
+    {
+        private string _filePath;
+
+        public EnvFileKeyValuePairs(string filePath)
+        {
+            _filePath = filePath;
+        }
+
+        public IEnumerator<KeyValuePair<string, string>> GetEnumerator()
+        {
+            var text = File.ReadAllText(_filePath) ?? string.Empty;
+            var index = 0;
+            var line = ReadLine(text, ref index);
+            while (line != null)
```

Comment:
This is just a copy paste with the key/output yields added and the error messages updated right? I wasn't able to find other changes

Reply:
Yup, that is correct :)